### PR TITLE
Swteam/#244/event show

### PR DESCRIPTION
**Pull request by fpottbaecker**

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -56,7 +56,7 @@ def deadline_has_passed?
   end
 
   def single_player?
-    player_type == Event.player_types[:single]
+    player_type == 'single'
```
Comment (ErikBrendel) [inline app/models/event.rb]: Is a string 'the way to go' when checking enum equality?

File: app/models/event.rb [hunk 2]
```diff
@@ -82,4 +82,22 @@ def can_leave?(user)
...
+    self.class.human_game_mode game_mode
+  end
+
+  class << self
+    def human_player_type(type)
+      I18n.t("activerecord.attributes.event.player_types.#{type}")
+    end
+
+    def human_game_mode(_)
+      'None'
```
Comment (ErikBrendel) [inline app/models/event.rb]: Can you somehow annotate that this method should get overridden by subclasses?

File: app/models/league.rb [hunk 3]
```diff
@@ -55,4 +55,8 @@ def round_robin_pairings(teams_array)
       [[teams_array.first, pivot]] + (1...(n / 2)).map { |j| [teams_array[j], teams_array[n - 1 - j]] }
     end
   end
+
+  def self.human_game_mode(mode)
+    I18n.t("activerecord.attributes.league.game_modes.#{mode}")
```
Comment (ErikBrendel) [inline app/models/league.rb]: Why did you change `game_modes` enum to singular, but use the plural in the translation key?

File: app/views/events/show.html.erb [hunk 4]
```diff
@@ -1,46 +1,68 @@
...
-        <% end %>
-      </span>
-    <% end %>
+    <%=t '.title', default: @event.model_name.human.titleize %>
   </h1>
+  <% if user_signed_in? %>
+    <span class="pull-right">
+      <% if current_user.organizing_events.include?(@event) %>
+        <% if can? :edit, @event %>
+          <%= link_to t('.edit', default: t("helpers.links.edit")),
```
Comment (ErikBrendel) [inline app/views/events/show.html.erb]: Why the double quotes here?

File: app/views/events/show.html.erb [hunk 5]
```diff
@@ -1,46 +1,68 @@
...
+    </span>
+  <% end %>
 </div>
 
 <dl class="dl-horizontal">
-  <% [:name, :description, :discipline, :game_mode, :max_teams, :deadline, :startdate, :enddate].each do |attribute| %>
-      <dt><strong><%= model_class.human_attribute_name(attribute) %>:</strong></dt>
-      <dd><%= @event.send attribute %></dd>
-  <% end %>
+  <dt><strong><%= model_class.human_attribute_name(:name) %>:</strong></dt>
```
Comment (ErikBrendel) [inline app/views/events/show.html.erb:26]: That really looks bad. Please do something about that duplication

File: features/step_definitions/navigation.rb [hunk 6]
```diff
@@ -5,3 +5,12 @@
 When /^he visits the new tournament path$/ do
   visit new_tournament_path
 end
+
+
```
Comment (ErikBrendel) [inline features/step_definitions/navigation.rb]: Space?

File: features/step_definitions/tournament.rb [hunk 7]
```diff
@@ -23,3 +23,8 @@
 When(/^the tournament overview page for (.*) is visited$/) do |tournamentName|
   visit event_overview_path (tournament_named tournamentName)
 end
+
+
```
Comment (ErikBrendel) [inline features/step_definitions/tournament.rb]: Space?

**Review by ErikBrendel — CHANGES_REQUESTED** [review]
A really nice PR size :)

File: app/models/event.rb (hunk 1 above)
Comment (fpottbaecker) [reply app/models/event.rb]: http://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html

According to docs, the enum attribute returns a string. Symbols actually result in false comparisons

File: app/models/league.rb (hunk 3 above)
Comment (fpottbaecker) [reply app/models/league.rb]: Because rails magic. The enums actually had the *wrong name the whole time*. However, rails automatically creates a corresponding plural method that returns all enum constants. Since this is the set of all game modes, i decide to use the plural for the translation keys

File: app/views/events/show.html.erb (hunk 4 above)
Comment (fpottbaecker) [reply app/views/events/show.html.erb]: I just changed indentation, will fix.

File: features/step_definitions/navigation.rb (hunk 6 above)
Comment (fpottbaecker) [reply features/step_definitions/navigation.rb]: the final frontier

File: app/views/events/show.html.erb (hunk 5 above)
Comment (fpottbaecker) [reply app/views/events/show.html.erb:26]: there is no nice way to do this. Views inherently have some amount of duplication, just look at the form fields for every model. The previous solution was arguably wrong, since it used `send` and couldn't handle injecting translations. I would ask you to provide a better solution.

File: features/step_definitions/tournament.rb (hunk 7 above)
Comment (fpottbaecker) [reply features/step_definitions/tournament.rb]: these are the voyages.

**Comment by ErikBrendel** [comment]
This pr seems to highly overlap with https://github.com/hpi-swt2/sport-portal/pull/264 - you authors should probably discuss this to prevent merge conflicts before they occur

File: app/models/event.rb (hunk 1 above)
Comment (ErikBrendel) [reply app/models/event.rb]: When I see correctly, there should be an `event.single?` method already

File: app/views/events/show.html.erb (hunk 5 above)
Comment (ErikBrendel) [reply app/views/events/show.html.erb:26]: Just define a function that takes two parameters: the attribute (`:name`) and the value (`@event.name`) and call it 9 times

**Review by ErikBrendel — APPROVED** [review]

File: app/models/event.rb (hunk 1 above)
Comment (fpottbaecker) [reply app/models/event.rb]: correct

File: app/views/events/show.html.erb (hunk 5 above)
Comment (fpottbaecker) [reply app/views/events/show.html.erb:26]: I'm not sure if you can just define a method like that in the respective helper...
Otherwise, it would still look ugly:
```erb
<% vals = {
  name: @event.name,
  discipline: @event.discipline,
  ...
  game_mode: @event.human_game_mode,
  ...
} %>
<% vals.each do |k, v| %>
  <dd><strong><%= model_class.human_attribute_name k %>:</strong></dd>
  <dt><%= v %></dt>
<% end %>
```

And thats not a nice piece of view code. I prefer the style of form views.
I would leave it as it is, since other loops depend on the fact that there will only be this format forever. This is the first time the previous pattern was broken (and i think it's not the last time).

**Comment by fpottbaecker** [comment]
Reviewed and accepted by @MerlindlH 

Pending because of conflicts with #264

**Comment by fpottbaecker** [comment]
This also resolves one criterium of #232

File: app/views/events/_form.html.erb [hunk 8]
```diff
@@ -35,7 +35,7 @@
   <div class="form-group">
     <%= f.label :player_type, class: 'control-label col-lg-2' %>
     <div class="col-lg-10">
-      <%= f.select(:player_type, Event.player_types.map {|k, v| [k.humanize, v]},
+      <%= f.select(:player_type, Event.player_types.map {|k, _| [Event.human_player_type(k), k]},
```
Comment (usame-algan) [inline app/views/events/_form.html.erb]: The values of the dropdown options should be 0, 1, 2, ... with the current enums that would be the value and not the key.

**Review by usame-algan — APPROVED** [review]

File: app/views/events/show.html.erb (hunk 5 above)
Comment (ErikBrendel) [reply app/views/events/show.html.erb:26]: Maybe you are right. Keep it like that if you like

**Review by ErikBrendel — APPROVED** [review]

File: features/events/show.feature [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    Given a tournament with gamemode ko
+    When the tournament's page is visited
+    Then the page header should show 'Tournament'
+    And the page should show 'K.-o.-System'
+    And the page should show 'Teamteilnahme'
+
+  Scenario: League
+    Given a league with gamemode round_robin
+    When the league's page is visited
+    Then the page header should show 'League'
```
Comment (Markus-Brand) [inline features/events/show.feature:13]: Why is "League" english but "Jeder gegen Jeden" and "Teamteilnahme" german? Please decide for one (English would fit the rest of the sentence)

**Review by Markus-Brand — APPROVED** [review]
lgtm, apart from the denglish cucumber tests
